drivers/gles3/storage/mesh_storage.cpp

```diff
@@ -1238,9 +1238,11 @@ void MeshStorage::_mesh_instance_remove_surface(MeshInstance *mi, int p_surface)
 
 	if (surface.vertex_buffers[0] != 0) {
 		GLES3::Utilities::get_singleton()->buffer_free_data(surface.vertex_buffers[0]);
-		GLES3::Utilities::get_singleton()->buffer_free_data(surface.vertex_buffers[1]);
 		surface.vertex_buffers[0] = 0;
-		surface.vertex_buffers[1] = 0;
+		if (surface.vertex_buffers[1] != 0) {
+			GLES3::Utilities::get_singleton()->buffer_free_data(surface.vertex_buffers[1]);
+			surface.vertex_buffers[1] = 0;
+		}
```

Comment:
This change should be reverted, and instead vertex_buffers should be renamed to `blend_shape_vertex_buffers` here. The `blend_shape_vertex_buffers` are always generated together so they can be freed together as well

Reply:
Yes. Maybe it intends to free `blend_shape_vertex_buffers`. I will have a look at the different about `vertex_buffers` and `blend_shape_vertex_buffers`.